### PR TITLE
Mageck test new submodule

### PR DESCRIPTION
**Pull request by LaurenceKuhl**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by LaurenceKuhl** [comment]
Hi, so I'm not too sure about this one, it's to implement https://sourceforge.net/p/mageck/wiki/usage/#count and I'm not sure about how to code  the either/or clause. So far the 2 flags are passed through the optional args, but if none of the 2 are provided it will cause the module to fail. If anyone has an example I could follow i wouldn't say no :D

File: modules/nf-core/mageck/test/main.nf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::mageck=0.5.9" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/mageck:0.5.9--py37h6bb024c_0':
+        'quay.io/biocontainers/mageck:0.5.9--py37h6bb024c_0' }"
+
+    input:
+
```
Comment (FriederikeHanssen) [inline modules/nf-core/mageck/test/main.nf:11]:
```suggestion
```

File: modules/nf-core/mageck/test/main.nf [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+
```
Comment (FriederikeHanssen) [inline modules/nf-core/mageck/test/main.nf:26]:
```suggestion
```

File: tests/config/pytest_modules.yml [hunk 3]
```diff
@@ -1,2805 +1,2809 @@
 abacas:
-  - modules/nf-core/abacas/**
```
Comment (FriederikeHanssen) [inline tests/config/pytest_modules.yml:2]: somehow you are adapting all of these lines. is that right

File: tests/modules/nf-core/mageck/test/main.nf [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+workflow test_mageck_test_day0_label {
+
+    input = [
+        [ id:'test', single_end:false ], // meta map
+        file(params.test_data['mus_musculus']['csv']['count_table'], checkIfExists: true)
+    ]
+
+
+    MAGECK_TEST ( input )
```
Comment (FriederikeHanssen) [inline tests/modules/nf-core/mageck/test/main.nf:27]: are you supposed to use the MAGECK_TEST_DAY=_LABEL anywhere?

File: tests/modules/nf-core/mageck/test/test.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+- name: mageck test test_mageck_test_day0_label
+  command: nextflow run ./tests/modules/nf-core/mageck/test -entry test_mageck_test_day0_label -c ./tests/config/nextflow.config -c ./tests/modules/nf-core/mageck/test/nextflow.config
+  tags:
+    - mageck
+    - mageck/test
+  files:
+    - path: output/mageck/test.gene_summary.txt
+      md5sum: 23d39164e44e7125a5f5912463fa5970
+    - path: output/mageck/test.sgrna_summary.txt
+      md5sum: 26e4a50272a0289b0261a5380276f273
```
Comment (FriederikeHanssen) [inline tests/modules/nf-core/mageck/test/test.yml:21]: can you add a check that the versions.yml exist?

**Review by FriederikeHanssen — COMMENTED** [review]
looks good, just some minor comments mostly on formatting :D

**Comment by FriederikeHanssen** [comment]
> Hi, so I'm not too sure about this one, it's to implement https://sourceforge.net/p/mageck/wiki/usage/#count and I'm not sure about how to code the either/or clause. So far the 2 flags are passed through the optional args, but if none of the 2 are provided it will cause the module to fail. If anyone has an example I could follow i wouldn't say no :D

I think this is not unusal behavoir for a lot of tools to fail when no parameter is provided. I also have a similar issue with controlfreec. Just providing them in the nextflow.config is sufficient for now. We are discussing this on a broader scale atm, to possible have docs on that in the meta.yml or so. Personally, I am a bit cautious about this, just because it will mean we will need to stay up to date on docs even more on each tool update

**Comment by LaurenceKuhl** [comment]
@nf-core-bot fix linting

**Pull request closed without merging**